### DISCUSSION TITLE
How to create a cumulative histogram in Hist?

### DISCUSSION THREAD

**Discussion by txell-1** [scikit-hep/hist#312] [Q&A] [answered]
Is there a given way to create a cumulative histogram using Hist? 
And also, is there a way to create this cumulative Histogram from a given 1d or 2d Hist object?
In Matplotlib it would be [this](https://www.geeksforgeeks.org/create-a-cumulative-histogram-in-matplotlib/), but in our case some of the functionalities available in Hist are needed.

**Comment by henryiii** [comment, marked as answer]
1D, anyway:

```python
array = np.cumsum(h.values())
```

If you want a histogram object out, then:

```python
cumsum_hist = h.copy()
cumsum_hist[...] = np.cumsum(h)
```


You can be explicit and use `h.values()`, but just `h` works inside cumsum, as it exposes the interface NumPy expects.

**Reply by txell-1** [reply]
Thank you for your help!
